### ISSUE TITLE
$ref at root level is not omitted

### ISSUE THREAD

**Issue by dibericky** [APIDevTools/json-schema-ref-parser#174]
**Steps to reproduce**

With a jsonSchema like:
```
{
  components: {
    schemas: {
      Pet: {
        type: "string"
      }
    }
  },
  $ref: "#/components/schemas/Pet"
};
```
the $ref is correctly resolved, but is kept a "$ref: "#" in the result.
[Example](https://codesandbox.io/s/bug-json-schema-ref-parser-uhjpk?file=/src/index.js:115-240)
```
+ expected - actual ... Lines skipped

  {
-   '$ref': '#',
    components: {
...
    type: 'string'
  }
```

**What is expected?**
I expect that the $ref is omitted from the result:

```
{
      components: {
        schemas: {
          Pet: {
            type: "string"
          }
        }
      },
      type: "string"
    }
```
**Other info**
- node version: v10.20.1
- json-schema-ref-parser version: 9.0.1

**Comment by Danielecina** [comment]
I have the same problem. 
This $ref left in the root, blocks the recursive resolution of the $ref

How to use example: 
* Fork example
* Open new terminal panel 
* run yarn test
* (Example)[https://codesandbox.io/s/bug-json-schema-ref-parser-8dehm?file=/src/index.js]

**Comment by philsturgeon** [comment]
Can you put the content in the root instead of defining it inside then hosting it with a $ref? This is known to work, and what you've got in this example isn't adding any benefit over it.

Also what is components, thats commonly used in OpenAPI v3.x which is a whole other thing.

Having siblings is not valid JSON Schema 07, it was added in JSON Schema 2019-09, which is being addressed in #145. Let's cover this there.
